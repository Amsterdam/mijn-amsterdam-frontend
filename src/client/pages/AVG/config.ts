import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
