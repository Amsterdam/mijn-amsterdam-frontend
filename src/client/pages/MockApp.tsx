import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
