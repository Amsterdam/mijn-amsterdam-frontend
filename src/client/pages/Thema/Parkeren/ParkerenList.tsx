import { useParams } from 'react-router';

import { ListPageParamKind } from './Parkeren-thema-config';
import { useParkerenData } from './useParkerenData.hook';
import { ListPagePaginated } from '../../../components/ListPagePaginated/ListPagePaginated';
import { useHTMLDocumentTitle } from '../../../hooks/useHTMLDocumentTitle';

export function ParkerenList() {
  const { kind = 'lopende-aanvragen' } = useParams<{
    kind: ListPageParamKind;
  }>();

  const {
    vergunningen,
    id: themaId,
    isLoading,
    isError,
    tableConfig,
    breadcrumbs,
    routeConfig,
  } = useParkerenData();
  useHTMLDocumentTitle(routeConfig.listPage);

  const { displayProps, title, filter, sort, listPageRoute } =
    tableConfig[kind];

  return (
    <ListPagePaginated
      items={vergunningen.filter(filter).sort(sort)}
      themaId={themaId}
      title={title}
      appRoute={listPageRoute}
      breadcrumbs={breadcrumbs}
      displayProps={displayProps}
      isLoading={isLoading}
      isError={isError}
    />
  );
}
