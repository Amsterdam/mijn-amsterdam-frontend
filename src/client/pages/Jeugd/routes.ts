import { JeugdDetail } from './JeugdDetail';
import { JeugdThemaPagina } from './JeugdThema';
import { FeatureToggle } from '../../../universal/config/feature-toggles';

const baseRoute = '/jeugd';

export const JeugdRoutes = [
  {
    route: baseRoute,
    Component: JeugdThemaPagina,
    isActive: FeatureToggle.zorgnedLeerlingenvervoerActive,
  },
  {
    route: `${baseRoute}/voorziening/:id`,
    Component: JeugdDetail,
    isActive: FeatureToggle.zorgnedLeerlingenvervoerActive,
  },
];
