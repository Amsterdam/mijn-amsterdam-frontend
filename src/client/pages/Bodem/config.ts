import { generatePath } from 'react-router-dom';

import { LoodMetingFrontend } from '../../../server/services/bodem/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';

const displayPropsAanvragen = {
  detailLinkComponent: 'Adres',
  datumAanvraagFormatted: 'Aangevraagd',
  status: 'Status',
};

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    ...tableConfigBase,
    title: 'Lopende aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => !bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
  },
  [listPageParamKind.completed]: {
    ...tableConfigBase,
    title: 'Afgehandelde aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.completed,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over lood in de bodem.',
    to: 'https://www.amsterdam.nl/wonen-leefomgeving/bodem/lood-grond/',
  },
] as const;
