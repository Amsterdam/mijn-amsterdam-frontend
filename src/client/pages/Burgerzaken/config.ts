import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { IdentiteitsbewijsFrontend, LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const displayPropsBase: DisplayProps<
  WithDetailLinkComponent<IdentiteitsbewijsFrontend>
> = {
  detailLinkComponent: 'Type',
  datumAfloopFormatted: 'Geldig tot',
};

const listPageParamKind = {
  identiteitsbewijzen: 'identiteitsbewijzen',
};

const displayProps = withOmitDisplayPropsForSmallScreens(displayPropsBase, [
  'datumAfloopFormatted',
]);

export const tableConfig = {
  [listPageParamKind.identiteitsbewijzen]: {
    title: 'Mijn reisdocumenten',
    displayProps,
    sort: dateSort('datumAfloop', 'desc'),
    listPageRoute: generatePath(AppRoutes['BURGERZAKEN/LIST'], {
      kind: listPageParamKind.identiteitsbewijzen,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/burgerzaken',
    title: 'Overzicht en aanvragen bij burgerzaken',
  },
] as const;

export const backLinkDetailPage = AppRoutes.BURGERZAKEN;
