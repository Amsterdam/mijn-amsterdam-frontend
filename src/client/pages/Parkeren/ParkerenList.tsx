import { useParams } from 'react-router-dom';

import { useParkerenData } from './useParkerenData.hook';
import { AppRoutes } from '../../../universal/config/routes';
import { Themas } from '../../../universal/config/thema';
import { ListPagePaginated } from '../../components/ListPagePaginated/ListPagePaginated';
import { ThemaTitles } from '../../config/thema';
import { ListPageParamKind } from '../VergunningenV2/config';

export function ParkerenList() {
  const params = useParams<{ kind: ListPageParamKind }>();

  const { parkeerVergunningen, isLoading, isError, tableConfig } =
    useParkerenData();
  const appRouteBack = AppRoutes.PARKEREN;

  const title = tableConfig[params.kind].title;
  const displayProps = tableConfig[params.kind].displayProps;

  return (
    <ListPagePaginated
      items={parkeerVergunningen}
      backLinkTitle={ThemaTitles.PARKEREN}
      title={title ?? ''}
      appRoute={AppRoutes['PARKEREN/LIST']}
      appRouteParams={params}
      appRouteBack={appRouteBack}
      displayProps={displayProps}
      thema={Themas.PARKEREN}
      isLoading={isLoading}
      isError={isError}
    />
  );
}
