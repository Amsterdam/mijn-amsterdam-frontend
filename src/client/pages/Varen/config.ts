import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { VarenFrontend } from '../../../server/services/varen/config-and-types';

export const displayPropsAanvragen = {
  detailLinkComponent: '',
  title: 'Omschrijving',
  dateRequestFormatted: 'Aangevraagd',
  status: 'Status',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (vergunning: VarenFrontend) => !vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (vergunning: VarenFrontend) => vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.completed,
    }),
    ...tableConfigBase,
  },
};
