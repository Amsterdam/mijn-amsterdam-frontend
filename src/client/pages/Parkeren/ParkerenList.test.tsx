import { render, screen, waitFor } from '@testing-library/react';
import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import { beforeAll, describe, expect, it, vi } from 'vitest';

import { AppRoutes } from '../../../universal/config/routes';
import { AppState } from '../../../universal/types';
import { CaseTypeV2 } from '../../../universal/types/vergunningen';
import { ThemaTitles } from '../../config/thema';
import MockApp from '../MockApp';
import { ParkerenList } from './ParkerenList';
import { appStateAtom } from '../../hooks/useAppState';
import { listPageParamKind } from '../VergunningenV2/config';

const testState = {
  PARKEREN: {
    content: {
      url: 'https://parkeervergunningen.amsterdam.nl/',
      isKnown: true,
    },
  },
  VERGUNNINGENv2: {
    status: 'OK',
    content: [
      {
        caseType: CaseTypeV2.GPP,
        dateDecision: null,
        dateDecisionFormatted: null,
        dateEnd: null,
        dateInBehandeling: null,
        dateInBehandelingFormatted: null,
        dateRequest: '2024-06-05T00:00:00',
        dateRequestFormatted: '05 juni 2024',
        dateStart: null,
        decision: null,
        fetchDocumentsUrl:
          'http://localhost:5000/api/v1/services/vergunningen/v2/962cCLy-d6nz4-85Cfyb2CaOKclPxVWCXF9L8T1lYamfgI25euHU1vf5OsA-qeyGYVuukIOquMqEFhww68MWxEW5LjLvu6jwplz4Hgs1LyE',
        id: 'Z-24-2233516',
        identifier: 'Z/24/2233516',
        key: 'D8DEC5AD3C6F456D954C53DEF791EAA3',
        link: {
          to: '/vergunningen/gpp/Z-24-2233516',
          title: 'Bekijk hoe het met uw aanvraag staat',
        },
        location: null,
        processed: false,
        status: 'open',
        title: 'Vergunning 1',
      },
      {
        caseType: CaseTypeV2.GPK,
        dateDecision: null,
        dateDecisionFormatted: null,
        dateEnd: null,
        dateInBehandeling: null,
        dateInBehandelingFormatted: null,
        dateRequest: '2024-06-05T00:00:00',
        dateRequestFormatted: '05 juni 2024',
        dateStart: null,
        decision: null,
        fetchDocumentsUrl:
          'http://localhost:5000/api/v1/services/vergunningen/v2/962cCLy-d6nz4-85Cfyb2CaOKclPxVWCXF9L8T1lYamfgI25euHU1vf5OsA-qeyGYVuukIOquMqEFhww68MWxEW5LjLvu6jwplz4Hgs1LyE',
        id: 'Z-24-2233517',
        identifier: 'Z/24/2233516',
        key: 'D8DEC5AD3C6F456D954C53DEF791EAA3',
        link: {
          to: '/vergunningen/gpp/Z-24-2233516',
          title: 'Bekijk hoe het met uw aanvraag staat',
        },
        location: null,
        processed: false,
        status: 'open',
        title: 'Vergunning 2',
      },
    ],
  },
} as unknown as AppState;

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('ParkerenList', () => {
  beforeAll(() => {
    (window.scrollTo as any) = vi.fn();
  });

  const routeEntry = generatePath(AppRoutes['PARKEREN/LIST'], {
    kind: listPageParamKind.inProgress,
  });

  const routePath = AppRoutes['PARKEREN/LIST'];

  function Component() {
    return (
      <MockApp
        routeEntry={routeEntry}
        routePath={routePath}
        component={ParkerenList}
        initializeState={initializeState}
      />
    );
  }

  it('should render the component and show the correct title', () => {
    render(<Component />);
    expect(screen.getAllByText(ThemaTitles.PARKEREN)[0]).toBeInTheDocument();
  });

  it('should display the list of parkeervergunningen', async () => {
    const screen = render(<Component />);
    expect(screen.asFragment()).toMatchSnapshot();

    await waitFor(() => {
      expect(screen.getByText('Vergunning 1')).toBeInTheDocument();
      expect(screen.getByText('Vergunning 2')).toBeInTheDocument();
    });
  });
});
