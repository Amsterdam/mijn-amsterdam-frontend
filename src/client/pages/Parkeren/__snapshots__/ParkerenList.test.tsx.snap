// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ParkerenList > should display the list of parkeervergunningen 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _OverviewPage_47d7c0"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="Parkeren"
          class="_ThemaIcon_2fc54d"
          fill="black"
          height="34px"
          viewBox="0 0 56 34"
          width="56px"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            transform="translate(0, -12.000000)"
          >
            <path
              d="M42,35 C44.8994949,35 47.25,37.3505051 47.25,40.25 C47.25,43.1494949 44.8994949,45.5 42,45.5 C39.1005051,45.5 36.75,43.1494949 36.75,40.25 C36.75,37.3505051 39.1005051,35 42,35 Z M12.25,35 C15.1494949,35 17.5,37.3505051 17.5,40.25 C17.5,43.1494949 15.1494949,45.5 12.25,45.5 C9.35050506,45.5 7,43.1494949 7,40.25 C7,37.3505051 9.35050506,35 12.25,35 Z M33.25,12.25 L45.5,24.5 L56,24.5 L56,40.25 L50.75,40.25 C50.75,35.4175084 46.8324916,31.5 42,31.5 C37.1675084,31.5 33.25,35.4175084 33.25,40.25 L21,40.25 C21,35.4175084 17.0824916,31.5 12.25,31.5 C7.41750844,31.5 3.5,35.4175084 3.5,40.25 L-9.9475983e-14,40.25 L-9.9475983e-14,24.5 L12.25,12.25 L33.25,12.25 Z M31.5,15.75 L22.75,15.75 L22.75,24.5 L40.1833496,24.5 L31.5,15.75 Z M19.25,15.75 L14,15.75 L5.25,24.5 L19.25,24.5 L19.25,15.75 Z"
              id="Combined-Shape"
            />
          </g>
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="ams-link ams-link--standalone _BackLink_3b337d"
          href="/"
        >
          <span
            class="ams-icon ams-icon--size-5"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              viewBox="0 0 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m22.857 32-16-16 16-16 2.9 2.91L12.677 16l13.08 13.09z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span>
            Terug
          </span>
        </a>
        <h2
          class="ams-heading ams-heading--level-1"
        >
          Lopende aanvragen
        </h2>
      </div>
    </header>
    <div
      class="ams-screen ams-screen--wide"
    >
      <div
        class="ams-grid"
      >
        <div
          class="ams-grid__cell ams-grid__cell--span-all"
        >
          <div
            class="ams-table"
          >
            <table
              class="ams-table__table _TableV2_4c6825"
            >
              <thead
                class="ams-table__header"
              >
                <tr
                  class="ams-table__row"
                >
                  <th
                    class="ams-table__header-cell"
                  >
                    Kenmerk
                  </th>
                  <th
                    class="ams-table__header-cell"
                  >
                    Soort vergunning
                  </th>
                  <th
                    class="ams-table__header-cell"
                  >
                    Aangevraagd
                  </th>
                </tr>
              </thead>
              <tbody
                class="ams-table__body"
              >
                <tr
                  class="ams-table__row"
                >
                  <td
                    class="ams-table__cell"
                  >
                    Z/24/2233516
                  </td>
                  <td
                    class="ams-table__cell"
                  >
                    <a
                      class="ams-link ams-link--standalone _MaRouterLink__no-underline-fat_aa758a"
                      href="/parkeren/gpp/Z-24-2233516"
                    >
                      Vergunning 1
                    </a>
                  </td>
                  <td
                    class="ams-table__cell"
                  >
                    05 juni 2024
                  </td>
                </tr>
                <tr
                  class="ams-table__row"
                >
                  <td
                    class="ams-table__cell"
                  >
                    Z/24/2233516
                  </td>
                  <td
                    class="ams-table__cell"
                  >
                    <a
                      class="ams-link ams-link--standalone _MaRouterLink__no-underline-fat_aa758a"
                      href="/parkeren/gpp/Z-24-2233516"
                    >
                      Vergunning 2
                    </a>
                  </td>
                  <td
                    class="ams-table__cell"
                  >
                    05 juni 2024
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
